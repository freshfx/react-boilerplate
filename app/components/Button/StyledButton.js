import styled from '@emotion/styled'

import buttonStyles from './buttonStyles'

const StyledButton = styled.button`
  ${buttonStyles};
`

export default StyledButton
