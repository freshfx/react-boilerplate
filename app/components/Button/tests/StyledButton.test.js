import 'jest-styled-components'
import React from 'react'
import {mount} from 'enzyme'

import StyledButton from '../StyledButton'

const renderComponent = (props = {}) => mount(<StyledButton {...props} />)

describe('<StyledButton />', () => {
  it('should match snapshot', () => {
    const renderedComponent = renderComponent()
    expect(renderedComponent).toMatchSnapshot()
  })
})
