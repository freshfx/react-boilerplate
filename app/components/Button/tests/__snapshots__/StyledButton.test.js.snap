// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledButton /> should match snapshot 1`] = `
.c0 {
  display: inline-block;
  box-sizing: border-box;
  padding: 0.25em 2em;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
  -webkit-font-smoothing: antialiased;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  outline: 0;
  font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: bold;
  font-size: 16px;
  border: 2px solid #41addd;
  color: #41addd;
}

.c0:active {
  background: #41addd;
  color: #fff;
}

<StyledButton>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "StyledButton-t20ntd-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "display:inline-block;box-sizing:border-box;padding:0.25em 2em;text-decoration:none;border-radius:4px;-webkit-font-smoothing:antialiased;-webkit-touch-callout:none;user-select:none;cursor:pointer;outline:0;font-family:'Helvetica Neue',Helvetica,Arial,sans-serif;font-weight:bold;font-size:16px;border:2px solid #41addd;color:#41addd;&:active{background:#41addd;color:#fff;}",
            ";",
          ],
        },
        "displayName": "StyledButton",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "StyledButton-t20ntd-0",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <button
      className="c0"
    />
  </StyledComponent>
</StyledButton>
`;
